### ISSUE TITLE
Question: How to re-use storageState so only have to log in once?

### ISSUE THREAD

**Issue by jonyo** [vitalets/playwright-bdd#77]
I see the example of how to set up auth context, I tried that.

The result: when I run, it goes through the log in steps for each browser.

What I'd like is to go through the login steps once, save the `storageState`, then have a `When I am logged in` (or a tag, either one) that would instruct it to use that storageState.

That way it only tries logging in once then re-uses the browser state across the rest of the tests for all browsers.

Is anything like that possible? or close to it?

**Comment by jzaratei** [comment]
I did that, using project dependencies
In your playwright.config.ts
```
{
    name: "LOGIN_CHROME",
    testDir: defineBddConfig({
      outputDir: ".features-gen/login",
      importTestFrom: "page-objects/basePage.ts",
      paths: ["features/login.feature"],
      require: ["steps/*.ts"],
    }),
    use: {
      ...devices["Desktop Chrome"],
  },
  {
    name: "REGRESSION_CHROME",
    testDir: defineBddConfig({
      outputDir: ".features-gen/regression",
      importTestFrom: "page-objects/basePage.ts",
      paths: ["features/regression/*.feature"],
      require: ["steps/*.ts"],
      verbose: true,
    });
    dependencies: ["SETUP_CHROME"],
    use: {
      ...devices["Desktop Chrome"],
      storageState: STORAGE_STATE,
    },
  },
```

That way everytime we trigger the regression test, the setup login project is executed first and store the session in storageState

**Comment by vitalets** [comment]
@jzaratei thank you 👍
@jonyo you can find more details on this approach in [Playwright docs](https://playwright.dev/docs/auth#basic-shared-account-in-all-tests). Basically, there are no specifics related to BDD, except that you can define authorization process with BDD steps and re-use them in auth-testing scenarios.

**Comment by jonyo** [comment]
Ah, yes that is what I already set up sans BDD so am familiar with it. I was hoping for a solution that somehow used the BDD tags to dictate that a specific test requires using the login state.

Perhaps I can use that in combo with setting the project's tags like `@authorized` for the one that requires the login, and `not @authorized` for the other...

Thanks!  👍

**Comment by QAPranav** [comment]
@jonyo hope you doing well can you please share your code how you used the set the storage state thank you

**Comment by jonyo** [comment]
> @jonyo hope you doing well can you please share your code how you used the set the storage state thank you

I just followed the [playwright docs](https://playwright.dev/docs/auth#basic-shared-account-in-all-tests) - the only thing different is that it has to be using bdd gen. So I made mine a simple "Given I log in for tests", made my step for "I log in for tests" and the rest follows the linked guide. Where at the bottom of the step file, after it is logged in, it saves the session to a file, then in the playwright config it uses that file.  (exactly what it says in the docs)

Hopefully that gets you started! 😄
